File: app/src/main/java/com/brooke/michael/filmbuff/ui/fragment/MainFragment.java
package com.brooke.michael.filmbuff.ui.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.astuetz.PagerSlidingTabStrip;
import com.brooke.michael.filmbuff.R;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Created by Michael on 06-Mar-16.
 */
public class MainFragment extends Fragment {

    @Bind(R.id.sliding_tabs)
    PagerSlidingTabStrip mSlidingTabs;

    @Bind(R.id.view_pager)
    ViewPager mViewPager;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, final ViewGroup container, Bundle savedInstanceState) {

        View rootView = inflater.inflate(R.layout.fragment_discover, container, false);

        ButterKnife.bind(this, rootView);

        MoviePagerAdapter adapter = new MoviePagerAdapter(getChildFragmentManager());
        mViewPager.setAdapter(adapter);
        mSlidingTabs.setViewPager(mViewPager);

        return rootView;
    }

    public class MoviePagerAdapter extends FragmentPagerAdapter {

        private final String[] TITLES = {
                "This Month",
                "Most Popular",
                "Highest Rated"
        };

        ThisMonthFragment thisMonthFragment;

        public MoviePagerAdapter(FragmentManager fm) {
            super(fm);
            thisMonthFragment = new ThisMonthFragment();
        }

        @Override
        public int getCount() {
            return TITLES.length;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return TITLES[position];
        }

        @Override
        public Fragment getItem(int position) {

            return new ThisMonthFragment();

//            Fragment frag = thisMonthFragment;
//
//            switch(position){
//                case 0:
//                    frag = thisMonthFragment;
//                    break;
//                case 1:
//                    frag = thisMonthFragment;
//                    break;
//                case 2:
//                    frag = thisMonthFragment;
//                    break;
//                default:
//                    frag = thisMonthFragment;
//            }
//
//            return frag;
        }

        @Override
        public int getItemPosition(Object object) {
            return POSITION_NONE;
        }

    }
}

File: app/src/main/java/com/brooke/michael/filmbuff/ui/fragment/MainFragment.kt
package com.brooke.michael.filmbuff.ui.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.brooke.michael.filmbuff.R

import kotlinx.android.synthetic.main.fragment_discover.*

class MainFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View?
            = inflater!!.inflate(R.layout.fragment_discover, container, false)

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        viewPager.offscreenPageLimit = 2
        viewPager.adapter = MoviePagerAdapter(childFragmentManager)
        slidingTabs.setViewPager(viewPager)
    }

    inner class MoviePagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        private val TITLES = arrayOf("This Month", "Most Popular", "Highest Rated")

        override fun getCount(): Int = TITLES.size

        override fun getPageTitle(position: Int): CharSequence = TITLES[position]

        override fun getItem(position: Int): Fragment = when(position){
            0 -> TabFragment(TabType.THIS_MONTH)
            1 -> TabFragment(TabType.MOST_POPULAR)
            2 -> TabFragment(TabType.HIGHEST_RATED)
            else -> TabFragment(TabType.THIS_MONTH)
        }

        override fun getItemPosition(`object`: Any?): Int = POSITION_NONE

    }
}
